leo_fw/scripts/firmware_parameter_bridge

```diff
@@ -0,0 +1,230 @@
+            .get_parameter_value()
+            .string_value
+        )
+
+        self.load_yaml_configs()
+        self.parse_yaml_configs()
+
+        param_client_cb_group = MutuallyExclusiveCallbackGroup()
+        self.firmware_parameter_service_client = self.create_client(
+            SetParameters,
+            "firmware/set_parameters",
+            callback_group=param_client_cb_group,
+        )
+
+        boot_client_cb_group = MutuallyExclusiveCallbackGroup()
+        self.frimware_boot_service_client = self.create_client(
+            Trigger,
+            "firmware/boot",
+            callback_group=boot_client_cb_group,
+        )
+
+        service_cb_group = MutuallyExclusiveCallbackGroup()
+        self.param_bridge_srv = self.create_service(
+            Trigger,
+            "upload_params",
+            self.param_bridge_srv_callback,
+            callback_group=service_cb_group,
+        )
+
+        sub_cb_group = MutuallyExclusiveCallbackGroup()
+        self.firmware_subscriber = self.create_subscription(
+            Empty,
+            "firmware/param_trigger",
+            self.param_bridge_sub_callback,
+            QoSProfile(
+                history=QoSHistoryPolicy.KEEP_LAST,
+                depth=1,
+                reliability=QoSReliabilityPolicy.BEST_EFFORT,
+                durability=QoSDurabilityPolicy.VOLATILE,
+            ),
+            callback_group=sub_cb_group,
+        )
+
+        self.get_logger().info("Starting node.")
+        self.send_params(boot_firmware=False)
+
+    def load_yaml_configs(self) -> None:
+        if self.default_param_file == "":
+            self.get_logger().error("Path to file with default parameters is empty!")
+            raise Exception()
+
+        with open(self.default_param_file, "r") as file:
+            try:
+                self.default_config: dict = yaml.safe_load(file)
+            except yaml.YAMLError as exc:
+                self.get_logger().error(exc)
+                raise Exception()
+
+        if self.override_param_file != "":
+            with open(self.override_param_file, "r") as file:
+                try:
+                    self.override_config: dict = yaml.safe_load(file)
+                except yaml.YAMLError as exc:
+                    self.get_logger().error(exc)
+                    raise Exception()
+        else:
+            self.get_logger().warning("Path to file with override parameters is empty.")
+
+    def parse_yaml_configs(
+        self,
+        param_name_prefix: str = "",
+        default_dict: dict = None,
+        override_dict: dict = None,
+    ) -> None:
+        if default_dict is None:
+            default_dict = self.default_config
+        if override_dict is None:
+            override_dict = self.override_config
+
+        default_keys = set(default_dict.keys())
+        override_keys = set(override_dict.keys())
+
+        for key in default_keys:
+            if type(default_dict[key]) == dict:
```

Comment:
Use isinstance() rather than type() for a typecheck
```suggestion
            if isinstance(default_dict[key], dict):
```

Reply:
Done